File: backend/src/main/java/com/drip/admin/modules/system/controller/UserController.java
package com.drip.admin.modules.system.controller;

import com.drip.admin.common.log.OperationLog;
import com.drip.admin.common.response.ApiResponse;
import com.drip.admin.common.response.PageResult;
import com.drip.admin.common.security.RequirePermission;
import com.drip.admin.modules.system.dto.*;
import com.drip.admin.modules.system.entity.SysUserEntity;
import com.drip.admin.modules.system.service.UserService;
import com.drip.admin.modules.system.vo.UserListVo;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.*;

import java.util.*;

import static com.drip.admin.shared.utils.AdminUtils.*;


@RestController
@RequestMapping("/system")
public class UserController {
    private final UserService userService;

   public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping("/user")
    @RequirePermission("system:user:list")
    public ApiResponse<PageResult<UserListVo>> users(@Valid UserQuery query) {
        return ApiResponse.success(userService.page(query));
    }

    @GetMapping("/user/{id}")
    @RequirePermission("system:user:detail")
    public ApiResponse<SysUserEntity> user(@PathVariable long id) {
        return ApiResponse.success(userService.detail(id));
    }

    @PostMapping("/user")
    @RequirePermission("system:user:create")
    @OperationLog(module = "用户管理", action = "新增用户")
    public ApiResponse<Long> createUser(@Valid @RequestBody UserSaveRequest request) {
        return ApiResponse.success(userService.create(request));
    }

    @PutMapping("/user/{id}")
    @RequirePermission("system:user:update")
    @OperationLog(module = "用户管理", action = "编辑用户")
    public ApiResponse<Void> updateUser(@PathVariable long id, @Valid @RequestBody UserSaveRequest request) {
        userService.update(id, request);
        return ApiResponse.success(null);
    }

    @DeleteMapping("/user/{id}")
    @RequirePermission("system:user:delete")
    @OperationLog(module = "用户管理", action = "删除用户")
    public ApiResponse<Void> deleteUser(@PathVariable long id) {
        userService.delete(id);
        return ApiResponse.success(null);
    }

    @PutMapping("/user/{id}/status")
    @RequirePermission("system:user:disable")
    @OperationLog(module = "用户管理", action = "变更用户状态")
    public ApiResponse<Void> userStatus(@PathVariable long id, @Valid @RequestBody StatusUpdateRequest request) {
        userService.updateStatus(id, request.statusOrDefault());
        return ApiResponse.success(null);
    }

    @PostMapping("/user/{id}/unlock")
    @RequirePermission("system:user:unlock")
    @OperationLog(module = "用户管理", action = "解除登录锁定")
    public ApiResponse<Void> unlockUser(@PathVariable long id) {
        userService.unlockLogin(id);
        return ApiResponse.success(null);
    }

    @PutMapping("/user/{id}/role")
    @RequirePermission("system:user:assignRole")
    @OperationLog(module = "用户管理", action = "分配角色")
    public ApiResponse<Void> userRoles(@PathVariable long id, @Valid @RequestBody RoleAssignRequest request) {
        userService.assignRoles(id, request.getRoleIds());
        return ApiResponse.success(null);
    }

    @PostMapping("/user/{id}/resetPassword")
    @RequirePermission("system:user:resetPassword")
    @OperationLog(module = "用户管理", action = "重置密码")
    public ApiResponse<Void> resetPassword(@PathVariable long id, @Valid @RequestBody PasswordResetRequest request) {
        userService.resetPassword(id, request.passwordOrDefault());
        return ApiResponse.success(null);
    }
}

File: backend-python/app/modules/system/controller/user_controller.py
from datetime import datetime

from fastapi import APIRouter, Depends, Query, Request
from sqlalchemy.ext.asyncio import AsyncSession

from app.common.api_response import ApiResponse, success
from app.common.pagination import PageQuery
from app.modules.system.dto.user_request import PasswordResetRequest, RoleAssignRequest, StatusUpdateRequest, UserSaveRequest
from app.modules.system.router import get_db, require_permission
from app.modules.system.service.user_service import UserService

router = APIRouter(tags=["user"])


def get_user_service(request: Request, db: AsyncSession = Depends(get_db)) -> UserService:
    return UserService(db, request.app.state.redis)


@router.get("/user", response_model=ApiResponse)
async def list_users(
    page: PageQuery = Depends(),
    username: str | None = None,
    real_name: str | None = Query(default=None, alias="realName"),
    phone: str | None = None,
    status: int | None = None,
    dept_id: int | None = Query(default=None, alias="deptId"),
    role_id: int | None = Query(default=None, alias="roleId"),
    created_from: datetime | None = Query(default=None, alias="createdFrom"),
    created_to: datetime | None = Query(default=None, alias="createdTo"),
    _: dict = Depends(require_permission("system:user:list")),
    service: UserService = Depends(get_user_service),
) -> ApiResponse:
    return success(
        await service.list_users(
            page,
            username,
            real_name,
            phone,
            status,
            dept_id,
            role_id,
            created_from,
            created_to,
        )
    )


@router.get("/user/{id}", response_model=ApiResponse)
async def user_detail(
    id: int, _: dict = Depends(require_permission("system:user:detail")), service: UserService = Depends(get_user_service)
) -> ApiResponse:
    return success(await service.detail(id))


@router.post("/user", response_model=ApiResponse)
async def create_user(
    body: UserSaveRequest, _: dict = Depends(require_permission("system:user:create")), service: UserService = Depends(get_user_service)
) -> ApiResponse:
    return success(await service.create(body))


@router.put("/user/{id}", response_model=ApiResponse)
async def update_user(
    id: int, body: UserSaveRequest, _: dict = Depends(require_permission("system:user:update")),
    service: UserService = Depends(get_user_service),
) -> ApiResponse:
    await service.update(id, body)
    return success()


@router.delete("/user/{id}", response_model=ApiResponse)
async def delete_user(
    id: int, session: dict = Depends(require_permission("system:user:delete")), service: UserService = Depends(get_user_service)
) -> ApiResponse:
    await service.delete(int(session["userId"]), id)
    return success()


@router.put("/user/{id}/status", response_model=ApiResponse)
async def user_status(
    id: int, body: StatusUpdateRequest, session: dict = Depends(require_permission("system:user:disable")),
    service: UserService = Depends(get_user_service),
) -> ApiResponse:
    await service.update_status(int(session["userId"]), id, body)
    return success()


@router.post("/user/{id}/unlock", response_model=ApiResponse)
async def unlock_user(
    id: int, _: dict = Depends(require_permission("system:user:unlock")), service: UserService = Depends(get_user_service)
) -> ApiResponse:
    await service.unlock(id)
    return success()


@router.put("/user/{id}/role", response_model=ApiResponse)
async def assign_user_roles(
    id: int, body: RoleAssignRequest, session: dict = Depends(require_permission("system:user:assignRole")),
    service: UserService = Depends(get_user_service),
) -> ApiResponse:
    await service.assign_roles(int(session["userId"]), id, body)
    return success()


@router.post("/user/{id}/resetPassword", response_model=ApiResponse)
async def reset_password(
    id: int, body: PasswordResetRequest, session: dict = Depends(require_permission("system:user:resetPassword")),
    service: UserService = Depends(get_user_service),
) -> ApiResponse:
    await service.reset_password(int(session["userId"]), id, body)
    return success()
